File: backend/friends.py
from flask import Blueprint

from . import friend_service

friends_bp = Blueprint('friends', __name__)


@friends_bp.route('/api/friends/summary', methods=['GET'])
def get_friend_summary():
    return friend_service.get_friend_summary()


@friends_bp.route('/api/friends/requests', methods=['POST'])
def create_friend_request():
    return friend_service.create_friend_request()


@friends_bp.route('/api/friends/requests/<int:request_id>/respond', methods=['POST'])
def respond_friend_request(request_id):
    return friend_service.respond_friend_request(request_id)


@friends_bp.route('/api/friends/messages', methods=['POST'])
def send_friend_message():
    return friend_service.send_friend_message()


@friends_bp.route('/api/friends/read', methods=['POST'])
def mark_friend_items_read():
    return friend_service.mark_friend_items_read()

File: src/lib/friends.js
import { authFetch } from './authFetch.js';

const parseJson = async (response) => {
  const payload = await response.json().catch(() => ({}));
  if (!response.ok) {
    throw new Error(String(payload?.error || payload?.message || 'Request failed').trim());
  }
  return payload;
};

export async function fetchFriendSummary(authToken = '') {
  const response = await authFetch('/api/friends/summary', {}, { authToken });
  return parseJson(response);
}

export async function sendFriendRequest(payload = {}, authToken = '') {
  const response = await authFetch('/api/friends/requests', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload),
  }, { authToken });
  return parseJson(response);
}

export async function respondFriendRequest(requestId, action, authToken = '') {
  const response = await authFetch(`/api/friends/requests/${encodeURIComponent(requestId)}/respond`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ action }),
  }, { authToken });
  return parseJson(response);
}

export async function sendFriendMessage(recipientUsername, body, authToken = '') {
  const response = await authFetch('/api/friends/messages', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ recipient_username: recipientUsername, body }),
  }, { authToken });
  return parseJson(response);
}

export async function markFriendItemsRead(payload = {}, authToken = '') {
  const response = await authFetch('/api/friends/read', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload),
  }, { authToken });
  return parseJson(response);
}
